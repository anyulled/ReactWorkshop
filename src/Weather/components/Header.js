// @flow
/**
 * Created by alrs on 11/07/2017.
 */
import * as React from "react";
import PropTypes from "prop-types";
import {PageHeader, Glyphicon} from "react-bootstrap";

type Props = {
    city:string
};

class Header extends React.Component<Props> {
    render(): React.Node {
        const {props: {city}} = this;
        return (<PageHeader><Glyphicon glyph="grain"/> {city || "Weather App" }</PageHeader>);
    }
}
Header.propTypes = {
    city: PropTypes.string.isRequired
};

export default Header;