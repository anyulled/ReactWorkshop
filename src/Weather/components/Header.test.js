import React from "react";
import Header from "./Header";
import {shallow} from "enzyme";

describe("Header", () => {
    it("Renders properly with props", () => {
        const city = "Barcelona";
        const wrapper = shallow(<Header city={city}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("PageHeader").childAt(2).text()).toContain(city);
    });

    it("Renders properly without props", () => {
        const city = undefined;
        const wrapper = shallow(<Header city={city}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("PageHeader").childAt(2).text()).toContain("Weather App");
    });
});