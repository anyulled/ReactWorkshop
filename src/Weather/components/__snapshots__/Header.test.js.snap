// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders properly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    city="Barcelona"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bsClass": "page-header",
      "children": Array [
        <Glyphicon
          bsClass="glyphicon"
          glyph="grain"
        />,
        " ",
        "Barcelona",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "glyphicon",
          "glyph": "grain",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
      "Barcelona",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bsClass": "page-header",
        "children": Array [
          <Glyphicon
            bsClass="glyphicon"
            glyph="grain"
          />,
          " ",
          "Barcelona",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "glyphicon",
            "glyph": "grain",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        "Barcelona",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header Renders properly without props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    city={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bsClass": "page-header",
      "children": Array [
        <Glyphicon
          bsClass="glyphicon"
          glyph="grain"
        />,
        " ",
        "Weather App",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "glyphicon",
          "glyph": "grain",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
      "Weather App",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bsClass": "page-header",
        "children": Array [
          <Glyphicon
            bsClass="glyphicon"
            glyph="grain"
          />,
          " ",
          "Weather App",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "glyphicon",
            "glyph": "grain",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
        "Weather App",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
